sieve/bc_eval.c

```diff
@@ -2567,7 +2572,29 @@ int sieve_eval_bc(sieve_execute_t *exe, int *impl_keep_p, sieve_interp_t *i,
 
         if (i->edited_headers) i->getheadersection(m, &headers);
 
-        res = do_keep(i, sc, actions, actionflags, headers);
+        if (i->ikeep.target) {
```

Comment:
I think with fileinto, we prefer mailboxid, then we try specialuse, then we try the mailbox itself.

Here, it looks like we can only ever specify one of those? Should we allow all 3 and follow the same logic?

Reply:
Rik led me to believe that we would know the mailbox before creating the script but I can extend this to be more like fileinto